content/modules/flow-ui/pages/views/views.adoc

```diff
@@ -0,0 +1,402 @@
+If the `View.BeforeShowEvent` listener contains logic to add components or load data, it will be done performed, which could lead to adding duplicate components or reloading data.
+
+[[ready-event]]
+=== ReadyEvent
+
+`ReadyEvent` is the last event (after `View.BeforeShowEvent`) in the view opening process. In this event listener, you can make final configuration of the view according to loaded data and show notifications or dialogs.
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=notifications-bean;ready-event]
+----
+
+NOTE: Consequent navigation to the same view that is currently open, leads to triggering `View.ReadyEvent` again for the same instance of the view. For example, the user navigates to a view for the first time, then a view instance is created and `View.ReadyEvent` triggers. Then, the user clicks on the same menu item, navigating to the same opened view again. We have the same view instance, but `View.ReadyEvent` triggers again. +
+If the `View.ReadyEvent` listener contains logic to add components or load data, it will be done performed, which could lead to adding duplicate components or reloading data.
+
+IMPORTANT: Both `BeforeShowEvent` and `ReadyEvent` work this way, because in case of navigation, they are fired by Vaadin's `BeforeEnterEvent` and `AfterNavigationEvent` lifecycle event handlers respectively.
+
+[[attach-event]]
+=== AttachEvent
+
+`AttachEvent` is fired after a view is attached to the UI.
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=attach-event]
+----
+
+NOTE: An event is fired by Vaadin, since a view is a UI component too.
+
+[[before-close-event]]
+=== BeforeCloseEvent
+
+`BeforeCloseEvent` is the first event in the view closing process. The view is still displayed and fully functional. In this event listener, you can check any conditions and prevent closing using the `preventClose()` method of the event.
+
+[source,java,indent=0]
+----
+include::example$/onboarding/src/main/java/com/company/onboarding/view/user/UserListEventsView.java[tags=before-close-event]
+----
+
+NOTE: In case of navigation, you need to postpone it too. Check the type of action and if it's `NavigateCloseAction`, then obtain Vaadin's `BeforeLeaveEvent` and call its `postpone()` method. A more complex example that considering both *dialog* mode and *navigation* can be found in the `io.jmix.flowui.view.StandardDetailView#preventUnsavedChanges` method which handles the default `preventUnsavedChanges` logic.
```

Comment:
Not clear to me: "In case of navigation, you need to postpone it too."
What I need to postpone and when?
Perhaps something like "If you prevent closing the view, in case of navigation you need to postpone it too."